smartsim/_core/generation/operations/ensemble_operations.py

```diff
@@ -0,0 +1,165 @@
+    """Ensemble Configure Operation"""
+
+    def __init__(
+        self,
+        src: pathlib.Path,
+        file_parameters: t.Mapping[str, t.Sequence[str]],
+        dest: t.Optional[pathlib.Path] = None,
+        tag: t.Optional[str] = None,
+    ) -> None:
+        """Initialize a EnsembleConfigureOperation
+
+        :param src: Path to source
+        :param file_parameters: File parameters to find and replace
+        :param dest: Path to destination
+        :param tag: Tag to use for find and replacement
+        """
+        check_src_and_dest_path(src, dest)
+        self.src = src
+        """Path to source"""
+        self.dest = dest
+        """Path to destination"""
+        self.file_parameters = file_parameters
+        """File parameters to find and replace"""
+        self.tag = tag if tag else default_tag
+        """Tag to use for the file"""
+
+
+EnsembleGenerationProtocolT = t.TypeVar(
+    "EnsembleGenerationProtocolT", bound=EnsembleGenerationProtocol
+)
+
+
+@dataclass
+class EnsembleFileSysOperationSet:
+    """Dataclass to represent a set of Ensemble file system operation objects"""
+
+    operations: list[EnsembleGenerationProtocol] = field(default_factory=list)
+    """Set of Ensemble file system objects that match the EnsembleGenerationProtocol"""
+
+    def add_copy(
+        self, src: pathlib.Path, dest: t.Optional[pathlib.Path] = None
+    ) -> None:
+        """Add a copy operation to the operations list
+
+        :param src: Path to source
+        :param dest: Path to destination
+        """
+        self.operations.append(EnsembleCopyOperation(src, dest))
+
+    def add_symlink(
+        self, src: pathlib.Path, dest: t.Optional[pathlib.Path] = None
+    ) -> None:
+        """Add a symlink operation to the operations list
+
+        :param src: Path to source
+        :param dest: Path to destination
+        """
+        self.operations.append(EnsembleSymlinkOperation(src, dest))
+
+    def add_configuration(
+        self,
+        src: pathlib.Path,
+        file_parameters: t.Mapping[str, t.Sequence[str]],
+        dest: t.Optional[pathlib.Path] = None,
+        tag: t.Optional[str] = None,
+    ) -> None:
+        """Add a configure operation to the operations list
+
+        :param src: Path to source
+        :param file_parameters: File parameters to find and replace
+        :param dest: Path to destination
+        :param tag: Tag to use for find and replacement
+        """
+        self.operations.append(
+            EnsembleConfigureOperation(src, file_parameters, dest, tag)
+        )
+
+    @property
+    def copy_operations(self) -> list[EnsembleCopyOperation]:
+        """Property to get the list of copy files.
+
+        :return: List of EnsembleCopyOperation objects
+        """
+        return self._filter(EnsembleCopyOperation)
+
+    @property
+    def symlink_operations(self) -> list[EnsembleSymlinkOperation]:
+        """Property to get the list of symlink files.
```

Comment:
1. omit property.
2. replace files with `operations` to be explicit that the files specified therein are not returned

```suggestion
        """Retrieve the list of symlink operations.
```